server/src/models/database/socobo-user.ts

```diff
@@ -1,16 +1,64 @@
+import { IsEmail, IsNotEmpty, MinLength, ValidateIf } from "class-validator";
 import { ProviderType } from "./../enums/provider-type";
 import { Role } from "./../enums/role";
+import { ValidationGroup } from "./../enums/validation-group";
 
 export class SocoboUser {
+
   public id: number;
+
+  @ValidateIf((o) => o.email === "", {
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @IsNotEmpty({
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @MinLength(5, {
+    groups: [ ValidationGroup.LOGIN ]
+  })
   public username: string;
+
+  @ValidateIf((o) => o.username === "", {
```

Comment:
Okay we can simply define this in the read me, but I thought this could maybe also be restricted directly by our api. Is there a easy way in to say that it is only valid to use one of the two? If not I think the current solution would be good.